prog/vnet/update_firewall_rules.rb

```diff
@@ -187,61 +187,28 @@ def before_run
       perm
     end
 
-    permissions.each do |perm|
-      aws_client.authorize_security_group_ingress({
-        group_id: vm.private_subnets.first.private_subnet_aws_resource.security_group_id,
-        ip_permissions: [perm]
-      })
-    rescue Aws::EC2::Errors::InvalidPermissionDuplicate
-      next
-    end
-
-    hop_remove_aws_old_rules
-  end
-
-  label def remove_aws_old_rules
-    rules = vm.firewalls.map(&:firewall_rules).flatten
-    ip4_rules = rules.select { !it.ip6? && it.port_range }
-    ip6_rules = rules.select { it.ip6? && it.port_range }
-
-    # Fetch existing security group rules
+    security_group_id = vm.private_subnets.first.private_subnet_aws_resource.security_group_id
     security_group = aws_client.describe_security_groups({
-      group_ids: [vm.private_subnets.first.private_subnet_aws_resource.security_group_id]
+      group_ids: [security_group_id]
     }).security_groups.first
 
     # Remove existing rules that aren't in our current rules list
-    permissions_to_revoke = security_group.ip_permissions.select do |permission|
-      permission.ip_protocol == "tcp" && # Skip if not TCP for now
-        ip4_rules.none? { |r|
-          r.cidr.to_s == permission.ip_ranges.first&.cidr_ip &&
-            r.port_range.begin == permission.from_port &&
-            r.port_range.end - 1 == permission.to_port
-        } &&
-        ip6_rules.none? { |r|
-          r.cidr.to_s == permission.ipv_6_ranges.first&.cidr_ipv_6 &&
-            r.port_range.begin == permission.from_port &&
-            r.port_range.end - 1 == permission.to_port
-        }
-    end.map! do |permission|
-      perm = {
-        ip_protocol: "tcp",
-        from_port: permission.from_port,
-        to_port: permission.to_port
-      }
-      perm[:ip_ranges] = permission.ip_ranges if permission.ip_ranges.any?
-      perm[:ipv_6_ranges] = permission.ipv_6_ranges if permission.ipv_6_ranges.any?
-      perm
+    security_group.ip_permissions.each do |permission|
+      aws_client.revoke_security_group_ingress({
```

Comment:
Doesn't removing all permissions and then adding new permissions, means that we are not allowing new connections momentarily, even from the allowed IP addresses? Also this is not atomic right, so if something fails in between that short time can be even longer (until we try it next time).

I think we need to compare the current allowed IPs with the target and decide what to do with each rule (add, remove, skip).